pilot/pkg/features/pilot.go

```diff
@@ -676,6 +676,9 @@ var (
 	// User should not rely on builtin resource labels, this flag will be removed in future releases(1.20).
 	EnableOTELBuiltinResourceLables = env.Register("ENABLE_OTEL_BUILTIN_RESOURCE_LABELS", false,
 		"If enabled, envoy will send builtin lables(e.g. node_name) via OTel sink.").Get()
+
+	EnableGatewayPolicyAttachmentOnly = env.Register("ENABLE_GATEWAY_POLICY_ATTACHMENT_ONLY", false,
```

Comment:
we donot need such a flag, as the targetRef is optional.

Reply:
The flag is for whether or not Gateway API gateways will ignore policies NOT attached via a targetRef (according to the Gateway API spec). This semantic change would break existing users so I hid it behind a feature flag

Reply:
You mean previous users have pa with workload selector, which is applied to waypoint?

If so, since ambient is alpha, we may not need this flag

Reply:
This isn't just for waypoints. TargetRef binding can be used for gateway API gateways too